mysqld_exporter.go

```diff
@@ -1608,6 +1633,62 @@ func scrapeTableStat(db *sql.DB, ch chan<- prometheus.Metric) error {
 	return nil
 }
 
+func scrapeQueryResponseTime(db *sql.DB, ch chan<- prometheus.Metric) error {
+	var queryStats uint8
+	err := db.QueryRow(queryResponseCheckQuery).Scan(&queryStats)
+	if err != nil {
+		log.Debugln("Query response time distribution is not present.")
+		return nil
+	}
+	if queryStats == 0 {
+		log.Debugln("query_response_time_stats is OFF.")
+		return nil
+	}
+
+	queryDistributionRows, err := db.Query(queryResponseTimeQuery)
+	if err != nil {
+		return err
+	}
+	defer queryDistributionRows.Close()
+
+	var (
+		length       string
+		count        uint64
+		total        string
+		histogramCnt uint64
+		histogramSum float64
+		countBuckets = map[float64]uint64 {}
+	)
+
+	for queryDistributionRows.Next() {
+		err = queryDistributionRows.Scan(
+			&length,
+			&count,
+			&total,
+		)
+		if err != nil {
+			return err
+		}
+
+		length = strings.TrimSpace(length)
+		le, _ := strconv.ParseFloat(length, 64)
+		total, _ := strconv.ParseFloat(strings.TrimSpace(total), 64)
+		countBuckets[le] = count
+		histogramCnt += count
+		histogramSum += total
+		ch <- prometheus.MustNewConstMetric(
```

Comment:
For completeness, you should export a +Inf entry with histogramSum as the value.